core/common/src/serializers/DateTimeUnitSerializers.kt

```diff
@@ -48,12 +47,12 @@ public object TimeBasedDateTimeUnitSerializer: KSerializer<DateTimeUnit.TimeBase
                             seen = true
                         }
                         CompositeDecoder.DECODE_DONE -> break@loop // https://youtrack.jetbrains.com/issue/KT-42262
-                        else -> throw UnknownFieldException(elementIndex)
+                        else -> throwUnknownIndexException(elementIndex)
                     }
                 }
             }
         }
-        if (!seen) throw MissingFieldException("nanoseconds")
+        if (!seen) throw MissingFieldException(missingField = "nanoseconds", serialName = descriptor.serialName)
```

Comment:
Explicit parameter names so one can be sure `missingField` and `serialName` (both are strings) are in the right order